src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -2567,7 +2567,11 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     // return actual source code.) So sourceFile has classfiles filtered out.
     final def sourceFile: AbstractFile = {
       val file = associatedFile
-      if ((file eq NoAbstractFile) || (file.path endsWith ".class")) null else file
+      if (
+        (file eq NoAbstractFile) || {
+          val path = file.path
+          path.endsWith(".class") || path.endsWith(".sig")
```

Comment:
Aka `FileUtils.endsClass(file.path)`.

But you'd have to copy/move that from src/compiler to src/reflect...